examples/storage/src/main.rs

```diff
@@ -13,91 +13,100 @@ use ariel_os::debug::{
         value
     } else {
-        info!("no counter value. first time running this test? try rebooting.");
+        info!("no counter value in storage. Is this the first time running this example?");
         0
     };
 
     if value > 10 {
-        info!("counter value > 10, aborting test to safe flash cycles.");
+        info!("counter value > 10, aborting test to save flash cycles");
         exit(ExitCode::SUCCESS);
     }
+    info!("");
 
     storage::insert("counter", value + 1).await.unwrap();
 
+    // By getting the Storage mutex directly, changing e.g., a counter,
+    // can be done atomically w.r.t. concurrent access from the same firmware:
     {
-        // By getting the Storage mutex directly, changing e.g., a counter,
-        // can be done atomically w.r.t. concurrent access from the same firmware:
         let mut s = storage::lock().await;
-        let value: Option<u32> = s.get("counter").await.unwrap();
+        let value: Option<u32> = s.get("atomic_counter").await.unwrap();
         let value = value.unwrap_or_default();
-        s.insert("counter", value + 1).await.unwrap();
+        s.insert("atomic_counter", value + 1).await.unwrap();
+        info!("Old atomic counter value at {}", value);
     }
+    info!("");
 
     // Storing a string value
     // For insertion, a literal can be used.
+    info!("Storing \"string_key\": \"string_value\" into storage");
     storage::insert("string_key", "string_value").await.unwrap();
 
-    // Getting a string value
+    // Retrieve a string value
     if let Some(string) = storage::get::<heapless::String<64>>("string_key")
         .await
         .unwrap()
     {
-        info!("got heapless string value {}", string);
+        info!("got heapless string value: \"{}\"", string);
     }
     if let Some(string) = storage::get::<arrayvec::ArrayString<64>>("string_key")
         .await
         .unwrap()
     {
         // no `defmt::Format` for arrayvec, so just print length
-        info!("got arrayvec string value with len {}", string.len());
+        info!("Attempting to retrieve string value as ArrayString: {:x}", string.as_bytes());
     }
+    info!("");
 
     // Storing an object
     let cfg = MyConfig {
-        val_uno: heapless::String::<64>::try_from("some value").unwrap(),
-        val_dos: 99,
+        val_one: heapless::String::<64>::try_from("some value").unwrap(),
+        val_two: 99,
     };
+    info!("Storing cfg object {} as struct", cfg);
     storage::insert("my_config", cfg).await.unwrap();
 
     // Getting an object
     // Type used for `get()` needs to match what was used for `insert()`.
     let cfg: Option<MyConfig> = storage::get("my_config").await.unwrap();
     if let Some(cfg) = cfg {
-        info!("got cfg {:?}", cfg);
+        info!("got cfg object: {:?}", cfg);
     }
 
     // Getting a value as raw bytes probably does not return what you want due
     // to the way postcard works
```

Comment:
When and how *does* it make sense to use that?

Reply:
cc @kaspar030

Reply:
hm. I used that as a test, and kept it in to showcase what can go wrong.
Maybe the whole section should be prefixed with `info!("See what happens, Larry:");`, and explain why and how the deserializing goes wrong the way it does. Maybe, keep only one of the error cases. Obviously, code comments don't cut it.

Reply:
maybe this just needs an `info!("Don't try this at home:");` ?

Reply:
"Don't try this at home" catches the spirit well.

Reply:
Maybe also "Note that this may break at any time, in particular when we start sorting that hash"

Reply:
Added a `info!` note

Reply:
That's added way up when there is the first mismatch -- is everything below "Don't try this in your code!" playing fast and lose?

Reply:
Nonetheless, it's an improvement, and it *does* point out the necessary things. Let's go and improve gradually (or skip it because the don't-try-at-home parts break once postcard hashes are in).